### PR TITLE
MARCS model reader and container

### PR DESCRIPTION
**Pull request by jvshields**
Type: 🚀 feature

This creates a class to hold the contents of a MARCS model file as downloaded from https://marcs.oreme.org/. 

There are separate methods to read the metadata of the model by matching with regular expressions, and then parse the model contents of the file into a pandas dataframe. The MARCS model class holds both of these objects. 

The model object is as general as possible, ingesting all metadata and the full contents of the file. Further processing to get ready for stardis, extracting only the parameters and contents needed to run stardis will be done separately. 

📌 Resources
https://marcs.oreme.org/ - Marcs model atmospheres used to initialize stardis

🚦 Testing
How did you test these changes?

[ ✅ ] Tested on a newly downloaded set of Marcs models

### :ballot_box_with_check: Checklist

- [x] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: stardis/io/model/MARCS.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    fname : string
+            Path to model file
+
+    Returns
+    -------
+    dict : dictionary
+            parameters of file
+    """
+
+    with gzip.open(fpath, "r") as file:
```
Comment (wkerzendorf) [inline stardis/io/model/MARCS.py:34]: `rb`

File: stardis/io/model/MARCS.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        contents = file.readlines(550)
+        lines = [line.decode() for line in contents]
+
+        # regex is used to search for the lines.
+        teff, teff_units = re.search(
```
Comment (wkerzendorf) [inline stardis/io/model/MARCS.py:39]: use match

File: stardis/io/model/MARCS.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+class MARCSModel(object):
+    """
+    Class to hold a MARCS model. Holds a dict of the meta information and a pandas dataframe of the contents.
+
+    """
+
+    meta: dict
+    data: pd.DataFrame
+
+
```
Comment (wkerzendorf) [inline stardis/io/model/MARCS.py:17]: make a constant that describes the data format

Comment (jvshields) [reply stardis/io/model/MARCS.py:18]: Code rewritten as described.

File: stardis/io/model/MARCS.py (hunk 1 above)
Comment (jvshields) [reply stardis/io/model/MARCS.py]: The problem was actually implicitly reading as 'rb'. Needed to use 'rt' instead for read in text mode.

**Review by sonachitchyan — DISMISSED** [review]
Looks good to me!

**Review by andrewfullard — CHANGES_REQUESTED** [review]
Needs tests!

**Comment by jvshields** [comment]
I added a test. I don't really know when to use np.testing.assert_almost_equal rather than assert. I'm guessing it's just when doing float comparison, but no calculations are run on any of the floats and they're not insanely precise, so I would figure floating point comparison wouldn't be an issue here.

**Review by andrewfullard — APPROVED** [review]

**Review by sonachitchyan — COMMENTED** [review]
Can you add the required keys that you're reading to the docsting? (Mayne to the class docstring?)

**Comment by jvshields** [comment]
> Can you add the required keys that you're reading to the docsting? (Mayne to the class docstring?)

Done.

**Review by sonachitchyan — APPROVED** [review]

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**